pkgs/servers/sql/postgresql/ext/pointcloud.nix

```diff
@@ -0,0 +1,172 @@
+{
+  autoconf,
+  automake,
+  cunit,
+  fetchFromGitHub,
+  installShellFiles,
+  lib,
+  libtool,
+  libxml2,
+  perl,
+  postgresql,
+  postgresqlBuildExtension,
+  postgresqlTestExtension,
+  postgresqlTestHook,
+  sphinx,
+  which,
+  zlib,
+}:
+postgresqlBuildExtension (finalAttrs: {
+  pname = "pointcloud";
+  version = "1.2.5";
+
+  outputs = [
+    "out"
+    "man"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "pgpointcloud";
+    repo = "pointcloud";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-uFbScxq21kt0jOjjyfMeO3i+bG2/kWS/Rrt3ZpOqEns=";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    installShellFiles
+    # for xml2-config
+    libxml2
+    libtool
+    perl
+    # for doc
+    sphinx
+    which
+  ];
+
+  buildInputs = [
+    zlib
+  ];
+
+  doCheck = true;
+
+  checkInputs = [
+    cunit
+  ];
+
+  nativeCheckInputs = [
+    postgresql
+    postgresqlTestHook
+  ];
+
+  preConfigure = ''
+    ./autogen.sh
+  '';
+
+  postInstall = ''
+    cd doc
+    make man
+    installManPage build/man/pgpointcloud.1
+  '';
+
+  passthru.tests = {
+    extension = postgresqlTestExtension {
+      inherit (finalAttrs) finalPackage;
+      # see https://pgpointcloud.github.io/pointcloud/concepts/schemas.html
+      withPackages = [ "postgis" ];
+      sql = ''
```

Comment:
What about moving it to separate .sql file?

Reply:
Good idea! So I moved everything in its own subfolder.